bundles/org.openhab.binding.mpd/src/main/java/org/openhab/binding/mpd/internal/protocol/MPDConnectionThread.java

```diff
@@ -0,0 +1,298 @@
+
+        this.listener = listener;
+
+        this.address = address;
+        this.port = port;
+        this.password = password;
+        setDaemon(true);
+    }
+
+    @Override
+    public void run() {
+        try {
+            while (!disposed.get()) {
+                try {
+                    synchronized (pendingCommands) {
+                        pendingCommands.clear();
+                        pendingCommands.add(new MPDCommand("status"));
+                        pendingCommands.add(new MPDCommand("currentsong"));
+                    }
+
+                    establishConnection();
+                    updateThingStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE, null);
+
+                    processPendingCommands();
+                } catch (IOException e) {
+                    updateThingStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                } catch (MPDException e) {
+                    updateThingStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+                }
+
+                isInIdle.set(false);
+                closeSocket();
+
+                if (!disposed.get()) {
+                    sleep(RECONNECTION_TIMEOUT_SEC * 1000);
+                }
+            }
+        } catch (InterruptedException e) {
+        }
+    }
+
+    /**
+     * dispose the connection
+     */
+    public void dispose() {
+        disposed.set(true);
+        Socket socket = this.socket;
+        if (socket != null) {
+            try {
+                socket.close();
+            } catch (IOException e) {
+            }
+            this.socket = null;
+        }
+    }
+
+    /**
+     * add a command to the pending commands queue
+     *
+     * @param command command to add
+     */
+    public void addCommand(MPDCommand command) {
+        insertCommand(command, -1);
+    }
+
+    private void insertCommand(MPDCommand command, int position) {
+        logger.debug("insert command '{}' at position {}", command.getCommand(), position);
+        int index = position;
+        synchronized (pendingCommands) {
+            if (index < 0) {
+                index = pendingCommands.size();
+            }
+            pendingCommands.add(index, command);
+            sendNoIdleIfInIdle();
+        }
+    }
+
+    private void updateThingStatus(ThingStatus status, ThingStatusDetail statusDetail, @Nullable String description) {
+        if (!disposed.get()) {
+            listener.updateThingStatus(status, statusDetail, description);
+        }
+    }
+
+    private void sendNoIdleIfInIdle() {
+        if (isInIdle.compareAndSet(true, false)) {
+            try {
+                sendCommand(new MPDCommand("noidle"));
+            } catch (IOException e) {
+                logger.debug("sendCommand(noidle) failed");
+            }
+        }
+    }
+
+    private void establishConnection() throws IOException, MPDException {
+        openSocket();
```

Comment:
Would it make sense to check to basic sanity of the configuration parameters before connecting? A malformed/empty host address or port is of course a configuration error.

Reply:
Done.